indico/modules/rb_new/client/js/modules/bookRoom/BookRoomModal.jsx

```diff
@@ -397,3 +400,55 @@ export default class BookRoomModal extends React.Component {
         );
     }
 }
+
+const mapStateToProps = (state, {roomId}) => {
+    const {
+        bookRoom: {filters: {recurrence, dates, timeSlot}, bookingForm: {timeline}},
+        user,
+    } = state;
+    const room = roomsSelectors.getRoom(state, {roomId});
+    return {
+        bookingData: {recurrence, dates, timeSlot},
+        availability: timeline && timeline.availability,
+        dateRange: timeline && timeline.dateRange,
+        user,
+        room,
+    };
+};
+
+const mapDispatchToProps = (dispatch) => ({
+    onSubmit: (
+        {reason, usage, user, isPrebooking},
+        {bookingData: {recurrence, dates, timeSlot}, room: roomData}) => {
+        return dispatch(actions.createBooking({
+            reason,
+            usage,
+            user,
+            recurrence,
+            dates,
+            timeSlot,
+            room: roomData,
+            isPrebooking
+        }));
+    },
+    dispatch
+});
+
+const mergeProps = (stateProps, dispatchProps, ownProps) => {
+    const {bookingData: filters, room} = stateProps;
+    const {dispatch} = dispatchProps;
```

Comment:
const {dispatch, ...realDispatchProps} = dispatchProps;

That way you avoid passing `dispatch` into the actual component props.